plume-models/src/blogs.rs

```diff
@@ -195,6 +195,23 @@ impl Blog {
         }
     }
 
+    pub fn find_by_host(c: &PlumeRocket, host: Host) -> Result<Blog> {
+        let from_db = blogs::table
+            .filter(blogs::custom_domain.eq(host))
+            .limit(1)
+            .load::<Blog>(&*c.conn)?
+            .into_iter()
+            .next();
+        if let Some(from_db) = from_db {
```

Comment:
you might wanna check [ok_or](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.ok_or)